File: api/src/main/java/com/kanban/api/model/UserSession.java
package com.kanban.api.model;

import com.kanban.api.config.authentication.UserPrincipal;
import lombok.Data;

@Data
public class UserSession {

	private final String firstName;
	private final String lastName;
	private final String username;
	private final String emailId;
	private final String token;


	public static UserSession create(final UserPrincipal user, final String token) {
		return new UserSession(user.getFirstName(), user.getLastName(), user.getUsername(), user.getEmail(), token);
	}

}

File: ui/src/app/model/UserSession.ts

export class UserSession {
  name: string;
  username: string;
  emailId: string;
  token: string;


  constructor(name: string, username: string, emailId: string, token: string) {
    this.name = name;
    this.username = username;
    this.emailId = emailId;
    this.token = token;
  }
}
